### PR TITLE
Issue #605: made get acceptable and required tokens abstract

### PR DESCRIPTION
**Pull request by rnveach**
Issue #605

`AbstractCheckTest.java` required one of it's tests to change.

### CODE REVIEW

**Comment by romani** [comment]
`AbstractJavadocCheck failed coverage check. Line coverage rate of 98.8% is below 100.0%`

details:
```
$ git diff
diff --git a/pom.xml b/pom.xml
index 354f846..7f06041 100644
--- a/pom.xml
+++ b/pom.xml
@@ -1502,7 +1502,7 @@
             <artifactId>cobertura-maven-plugin</artifactId>
             <version>2.7</version>
             <configuration>
-              <quiet>true</quiet>
+              <!--<quiet>true</quiet>-->
               <formats>


$ mvn verify
....
[INFO] --- cobertura-maven-plugin:2.7:check (default) @ checkstyle ---
[INFO] Cobertura 2.1.1 - GNU GPL License (NO WARRANTY) - See COPYRIGHT file
[INFO] Cobertura: Loaded information on 400 classes.
[ERROR] com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheck failed coverage check. Line coverage rate of 98.8% is below 100.0%

[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
...
```

**Comment by rnveach** [comment]
wercker:
> maven-checkstyle-plugin:2.17:check: java.lang.AbstractMethodError: null

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java [hunk 1]
```diff
@@ -229,6 +229,16 @@ public void leaveJavadocToken(DetailNode ast) {
         return new int[] {TokenTypes.BLOCK_COMMENT_BEGIN };
     }
 
+    @Override
+    public int[] getAcceptableTokens() {
```
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java]: Should these be final like `getDefaultTokens`?

Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java]: yes, the same principle.

**Comment by rnveach** [comment]
> AbstractJavadocCheck failed coverage check

Fixed with new test similar to AbstractCheck.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java (hunk 1 above)
Comment (rnveach) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/AbstractJavadocCheck.java]: Done.

**Comment by rnveach** [comment]
````
aused by: java.lang.AbstractMethodError
	at com.puppycrawl.tools.checkstyle.TreeWalker.validateDefaultTokens(TreeWalker.java:284)
	at com.puppycrawl.tools.checkstyle.TreeWalker.registerCheck(TreeWalker.java:211)
	at com.puppycrawl.tools.checkstyle.TreeWalker.setupChild(TreeWalker.java:165)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:171)
	at com.puppycrawl.tools.checkstyle.Checker.setupChild(Checker.java:452)
	at com.puppycrawl.tools.checkstyle.api.AutomaticBean.configure(AutomaticBean.java:171)
	at org.apache.maven.plugin.checkstyle.exec.DefaultCheckstyleExecutor.executeCheckstyle(DefaultCheckstyleExecutor.java:218)
	at org.apache.maven.plugin.checkstyle.CheckstyleViolationCheckMojo.execute(CheckstyleViolationCheckMojo.java:538)
	at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
	... 21 more
````

**Comment by rnveach** [comment]
@romani I haven't confirmed it yet as we don't have anything printing out, but they do use their own custom checks:
https://github.com/xwiki/xwiki-commons/blob/86ae9b25e49aad08dd00a93572ac0e7ba4033907/xwiki-commons-tools/xwiki-commons-tool-verification-resources/src/main/resources/checkstyle-blocker.xml#L7

**Pull request merged**